Parser/action_helpers.c

```diff
@@ -1444,8 +1444,16 @@ expr_ty _PyPegen_formatted_value(Parser *p, expr_ty expression, Token *debug, Re
         conversion_val = (int)'r';
     }
 
+    expr_ty format_expr = format ? (expr_ty) format->result : NULL;
+    if (format_expr && format_expr->kind == JoinedStr_kind && asdl_seq_LEN(format_expr->v.JoinedStr.values) == 1) {
```

Comment:
I don't understand this exactly. And it seems like it fails when the format specifier is not just an expression:

```python3
>>> import datetime
>>> y = 4
>>> datetime.datetime.now().__format__(f"h{y=}")
'hy=4'
>>> f"{datetime.datetime.now():h{y=}}"
'hdatetime.datetime.now()4'
```